subsys/bluetooth/services/fast_pair/fp_crypto/include/fp_crypto_pn.h

```diff
@@ -0,0 +1,100 @@
+/*
+ * Copyright (c) 2022 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#ifndef _FP_CRYPTO_PN_H_
+#define _FP_CRYPTO_PN_H_
+
+/**
+ * @defgroup fp_crypto_pn Fast Pair Personalized Name crypto
+ * @brief Internal API of Fast Pair Personalized Name crypto
+ *
+ * @{
+ */
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/** Length of nonce (64 bits = 8 bytes). */
+#define FP_CRYPTO_PN_NONCE_LEN			8U
+/** Length of Additional Data packet header (128 bits = 16 bytes). */
+#define FP_CRYPTO_PN_ADDITIONAL_DATA_HEADER_LEN	16U
+
+/** Encrypt data using AES-CTR.
+ *
+ * @param[out] out Buffer to receive encrypted data.
+ * @param[in] in Plaintext data.
+ * @param[in] data_len Length of input and output data (in bytes).
+ * @param[in] key 128-bit (16-byte) AES key.
+ * @param[in] nonce 64-bit (8-byte) nonce.
+ *
+ * @return 0 If the operation was successful. Otherwise, a (negative) error code is returned.
+ */
+int fp_crypto_aes_ctr_encrypt(uint8_t *out, const uint8_t *in, size_t data_len, const uint8_t *key,
```

Comment:
And those functions could end up in `fp_crypto_ext_priv.h` so that we can test it but don't expose it.

Reply:
Hmm, couldn't you include the source file (include "fp_crypto_pn.c:") in the test source file to avoid unnecessary private headers?

Reply:
Currently we are building crypto and storage as a separate library to avoid duplicating KConfig options. Then the libraries are linked. The `_priv` approach has been already used in Fast Pair storage. Thanks to this we can explicitle state what API is needed for tests and what API is is actually used by Fast Pair service.